src/test/java/roomescape/repository/H2DatabaseTest.java

```diff
@@ -0,0 +1,47 @@
+package roomescape.repository;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
```

Comment:
webEnvironment 필드는 보통 웹 요청을 받는 부분에서 선언해주는 편입니다. 어떤 필드인지 알아보시고, 어떤 역할을 하는지 확인해보세요. (서블릿 등의 개념이 등장하는데 너무 어렵다면 1보 후퇴하시거나, GPT에게 쉽게 설명해달라고 요청해보세요. 저는 Chat GPT가 다른 부분보다도 학습혁명이라고 생각해서 학습도우미로 사용하시기를 권합니다. 어려운 예시와 함께 쉽게 설명해주는 도사입니다)

Reply:
> By default, @SpringBootTest will not start a server. You can use the webEnvironment attribute of @SpringBootTest to further refine how your tests run:
MOCK(Default) : Loads a web ApplicationContext and provides a mock web environment. Embedded servers are not started when using this annotation. If a web environment is not available on your classpath, this mode transparently falls back to creating a regular non-web ApplicationContext. It can be used in conjunction with [@AutoConfigureMockMvc or @AutoConfigureWebTestClient](https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.testing.spring-boot-applications.with-mock-environment) for mock-based testing of your web application.
>
> RANDOM_PORT: Loads a WebServerApplicationContext and provides a real web environment. Embedded servers are started and listen on a random port.
>
> DEFINED_PORT: Loads a WebServerApplicationContext and provides a real web environment. Embedded servers are started and listen on a defined port (from your application.properties) or on the default port of 8080.
>
> NONE: Loads an ApplicationContext by using SpringApplication but does not provide any web environment (mock or otherwise).

[공식문서](https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.testing)에 따르면 `@SpringBootTest`는 기본적으로 서버를 시작하지 않습니다.
또한 공식문서에서는 추가적으로 webEnvironment 속성을 설정함으로써 실제 웹 서버를 띄우는 등 테스트가 돌아가는 방식을 정제할 수 있음을 소개하고 있습니다.

해당 H2 데이터베이스에 대한 테스트는 서버를 띄우고 웹요청을 처리해보는 맥락이 필요없습니다.
따라서 webEnviroment에 대한 설정이 불필요함을 알게 되었습니다.

불필요한 컨텍스트 잘라내는 개선 진행해보겠습니다 👍
다만 학습하면서 서블릿의 개념을 발견하지는 못했는데요..! 웨지가 의도하신 학습의 흐름이 아니었을까봐 걱정되는 부분이 있어요.
추가적으로 알아봐야 하는 키워드가 있을까요?!

Reply:
Gpt 활용법에 대한 추천 감사합니다 😊
확실히 gpt는 작업을 도와주는 것 보다 학습을 도와주는데에 훨씬 강점이 있는 것 같아요.

잘 활용해보겠습니다 💪🏻

Reply:
실제 주석 코드 들어가서 내려가다보면 WebEnvironment enum에 대해 발견할 수 있습니다.
여기에 servlet에 대한 언급이 있어서 말씀드려 보았습니당 ㅎ
<img width="699" alt="image" src="https://github.com/woowacourse/spring-roomescape-admin/assets/51393021/f2f42919-d51b-406e-bd62-3422cb73028e">

Reply:
> 해당 H2 데이터베이스에 대한 테스트는 서버를 띄우고 웹요청을 처리해보는 맥락이 필요없습니다.
따라서 webEnviroment에 대한 설정이 불필요함을 알게 되었습니다.

지금 학습하신 수준에서 충분한 것 같아요. 

나~중에 포트를 잡아주는 설정이 테스트에 존재하는 이유 같은 게 궁금해지면 또 돌아오시고요 ㅎ